concordium-node/src/consensus_ffi/ffi.rs

```diff
@@ -709,14 +714,37 @@ impl ConsensusContainer {
         ))
     }
 
-    pub fn in_baking_committee(&self) -> ConsensusIsInBakingCommitteeResponse {
-        wrap_c_committee_call!(self, |consensus| bakerIdBestBlock(consensus))
+    pub fn in_baking_committee(&self) -> (ConsensusIsInBakingCommitteeResponse, Option<u64>) {
+        let consensus = self.consensus.load(Ordering::SeqCst);
+
+        unsafe {
+            let mut baker_id: u64 = 0;
+            let mut has_baker_id: u8 = 0;
+
+            let result = bakerStatusBestBlock(consensus, &mut baker_id, &mut has_baker_id);
+            let status =
+                ConsensusIsInBakingCommitteeResponse::try_from(result).unwrap_or_else(|code| {
+                    panic!("Unknown Consensus Baking Committee FFI return code: {}", code)
+                });
+
+            let baker_id_option = match has_baker_id {
+                0 => None,
+                1 => Some(baker_id),
+                code => panic!("Expected value from FFI to be either 0 or 1, received {}", code),
+            };
+
+            (status, baker_id_option)
+        }
     }
 
     pub fn in_finalization_committee(&self) -> bool {
         wrap_c_bool_call!(self, |consensus| checkIfWeAreFinalizer(consensus))
     }
 
+    pub fn is_running(&self) -> bool {
```

Comment:
Please document what this is meant to be doing. `is_running` does not really say what it is that is supposed to be running.

Reply:
tried giving it a better name as well.